### PR TITLE
Add a log on catch(error) in cardano/js-estimateMaxSpendable.ts

### PR DESCRIPTION
**Pull request by gre**
### 📝 Description

code like

```js
  try {
    typhonTransaction = await buildTransaction(a as CardanoAccount, t);
  } catch (error) {
    return new BigNumber(0);
  }
```

must be improved, because this is literally silenting possible errors. Instead we must at least LOG them using ledgerhq/logs.

This PR solves this.

Important notes for Cardano maintainers
---

**(1)**

**⚠️  Current bug in estimateMaxSpendable implementation:**

Please also note that this helped identify a problem, it seems like this codebase always fails at this point with

```
info: Failed to estimate max spendableError: Invalid Address
```

to reproduce, you can simply try to estimate the max spendable of the bot account:

```
VERBOSE=1 pnpm run:cli estimateMaxSpendable --id js:2:cardano:da85e83cab046f8a50aa6fcd67bd7dffee0e1fd15ad3345a78838239d4d3fc2291f70ea636c86770da1ab8bf2136d60d3f9e49292a5f32d1f460bd4c7e251372:cardano
```

and even tho the account do have **8 ADA**, the estimation have a spendable returning ZERO.

**(2)**

⚠️ 🤔  Why are we even `try catching` the buildTransaction? This should also be documented in a js comment 🙏 

---

### ❓ Context

- **Impacted projects**: `n/a` 
- **Linked resource(s)**: `n/a` 

### ✅ Checklist

- [x] **Test coverage** 
- [x] **Atomic delivery** 
- [x] **No breaking changes** 

### 🚀 Expectations to reach

_Please make sure you follow these [**Important Steps**](https://github.com/LedgerHQ/ledger-live/blob/develop/CONTRIBUTING.md#important-steps)._

_Pull Requests must pass the CI and be internally validated in order to be merged._

### CODE REVIEW

File: libs/ledger-live-common/src/families/cardano/js-estimateMaxSpendable.ts [hunk 1]
```diff
@@ -42,6 +43,7 @@ const estimateMaxSpendable = async ({
   try {
     typhonTransaction = await buildTransaction(a as CardanoAccount, t);
   } catch (error) {
+    log("cardano-error", "Failed to estimate max spendable: " + String(error));
     return new BigNumber(0);
```
Comment (hedi-edelbloute) [inline libs/ledger-live-common/src/families/cardano/js-estimateMaxSpendable.ts:47]: Can we even continue the normal flow if we get in the catch clause ?

Comment (gre) [reply libs/ledger-live-common/src/families/cardano/js-estimateMaxSpendable.ts:47]: indeed, i wonder if it's necessary to keep this try-catch part. basically my `(2)` question in description for the cardano maintainers 🙏 
but if we keep it it must be both documented with a comment and we need to log the error as suggested here ;)

Comment (hedi-edelbloute) [reply libs/ledger-live-common/src/families/cardano/js-estimateMaxSpendable.ts:47]: Agree with you... should we create a ticket to try to remove the try catch thingie ? @haammar-ledger any thoughts ?

Comment (hedi-edelbloute) [reply libs/ledger-live-common/src/families/cardano/js-estimateMaxSpendable.ts:47]: Maybe we can first log the thing and then by this ticket try to investigate if we can remove the try catch safely

**Review by hedi-edelbloute — APPROVED** [review]

**Comment by haammar-ledger** [comment]
@pavanvora Interested in your thoughts on the issue raised in the description. Thanks. :slightly_smiling_face:

**Pull request merged**